flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -167,6 +168,8 @@
     private final ResourceCleaner localResourceCleaner;
     private final ResourceCleaner globalResourceCleaner;
 
+    private final AtomicBoolean shutDown = new AtomicBoolean(false);
```

Comment:
```suggestion
    private final volatile boolean isShutdown = false;
```
For primitive values like boolean, you could use the `volatile` keyword to mark a member for multi-threaded access. This will prevent the data to be loaded into the Threads stack. Threads will, instead, look it up directly on the heap. This is sufficient for primitive values like boolean, because they are written in a atomic fashion.

An example of such a usage can be found in [ClosingFSDataOutputStream.closed](https://github.com/apache/flink/blob/e0b4cbb5694d9c58f871b79e52da250a2f08cd2a/flink-core/src/main/java/org/apache/flink/core/fs/ClosingFSDataOutputStream.java#L39)

Reply:
Aha, right. I saw uses of `volatile boolean` and `AtomicBoolean` throughout, thanks for clarifying the usage.